apps/air-discount-scheme/web/screens/Subsidy/components/Benefits/Benefits.tsx

```diff
@@ -32,89 +37,92 @@ const FetchDiscountsMutation = gql`
 `
 
 function Benefits({ misc }: PropTypes) {
-  const [fetchDiscounts, { data }] = useMutation(FetchDiscountsMutation)
-  const { user: authUser } = useContext(UserContext)
+  const [fetchDiscounts, { data, loading, called }] = useMutation(
+    FetchDiscountsMutation,
+  )
   useEffect(() => {
     fetchDiscounts()
   }, [fetchDiscounts])
 
-  const { fetchDiscounts: codes } = data || {}
-  const {
-    myRights,
-    remaining,
-    copyCode,
-    codeDescription,
-    kidsRights,
-  } = JSON.parse(misc)
-
+  const { fetchDiscounts: discounts = [] } = data || {}
+  const { myRights, codeDescription, attention, codeDisclaimer } = JSON.parse(
+    misc,
+  )
+  const { activeCodes, fundUsed, noRights } = discounts.reduce(
```

Comment:
Don't see the point of mutably reducing this since it's just destructed when rendered.

Whole point is to find if the user has any benefits, rather do:
```typescript
const hasBenefits = discounts.some(({user}) =>
  user.meetsRequirement || user.fund.used === user.fund.total)
```

Reply:
the main point is to order by status, `[...activeCodes, ...fundUsed, ...noRights]` and add status code to each group

Reply:
Yeah I see, but the currently logged in user should always be on top.